plutus-core/plutus-ir/src/PlutusIR/TypeCheck/Internal.hs

```diff
@@ -219,21 +220,28 @@ ty ~> vTy
 -}
 inferTypeM (Let ann r@NonRec bs inTerm) = do
```

Comment:
What about recursive let bindings? We don't actually allow type bindings in such a position, but I think they should typecheck. Or at least let's test it.